moler/cmd/unix/nmap.py

```diff
@@ -55,14 +55,15 @@ def on_new_line(self, line, is_full_line):
                 self._parse_ports_line(line)
                 self._parse_raw_packets(line)
                 self._parse_scan_report(line)
+                self._parse_scan_reports(line)
                 self._parse_syn_stealth_scan(line)
                 self._parse_skipping_host(line)
             except ParsingDone:
                 pass
         return super(Nmap, self).on_new_line(line, is_full_line)
 
     _re_ports_line = re.compile(r"^(?P<LINES>(?P<PORTS>(?P<PORT>\d+)\/(?P<TYPE>\w+))\s+"
-                                r"(?P<STATE>\S+)\s+(?P<SERVICE>\S+)\s+(?P<REASON>\S+)\s*)$")
+                                r"(?P<STATE>\S+)\s+(?P<SERVICE>\S+)\s*(?P<REASON>\S+)?\s*)$")
```

Comment:
What separates 2 \S+ if separator changed from \s+ into \s*?

Reply:
Nothing - in this case group 'REASON' does not exists.
Example here (compare the last group with others) - https://regex101.com/r/IOsB5U/1